weave/stitch.py

```diff
@@ -114,17 +119,16 @@ def print_debug_summary(self) -> None:
             if recorder.tags:
                 res += (
                     "\n"
-                    + f"    tags: {','.join([str((tag_name, node_names[tag_recorder.node])) for tag_name, tag_recorder in recorder.tags.items()])}"
+                    + f"    tags: {','.join([str((tag_name, (node_names[tag_recorder.node] if tag_recorder.node is not None else '<None>'))) for tag_name, tag_recorder in recorder.tags.items()])}"
                 )
         print(res)
 
 
-def stitch(
-    leaf_nodes: list[graph.Node],
+def stitch_and_catch(
+    leaf_nodes: typing.Sequence[typing.Union[graph.Node, Exception]],
```

Comment:
seems odd to have `leaf_nodes` include `Exception`. The caller could just pass in the good nodes, then re-construct the list with inter-stitched exceptions. This would keep exceptions out of this code path.